public/class-paystack-forms-public.php

```diff
@@ -62,28 +62,82 @@ public function enqueue_scripts()
 
-function kkd_pff_paystack_add_paystack_charge($amount)
+class Kkd_Pff_Paystack_PaystackCharge
 {
-    // $amountinkobo = $amount; // * 100;
-    $charge = 0;
-    $amount = intval($amount);
-    if ($amount <= 2500) {
-        $charge = floatval($amount*KKD_PFF_PAYSTACK_PERCENTAGE);
-    } else {
-        $charge = floatval($amount*KKD_PFF_PAYSTACK_PERCENTAGE)+100;
+    public $percentage;
+    public $additional_charge;
+    public $crossover_total;
+    public $cap;
+
+    public $charge_divider;
+    public $crossover;
+    public $flatline_plus_charge;
+    public $flatline;
+
+    public function __construct($percentage = 0.015, $additional_charge = 10000, $crossover_total = 250000, $cap = 200000)
+    {
+        $this->percentage = $percentage;
+        $this->additional_charge = $additional_charge;
+        $this->crossover_total = $crossover_total;
+        $this->cap = $cap;
+        $this->__setup();
     }
-    // echo $charge;
-    if ($charge > 2000) {
-        $charge = 2000;
+
+    private function __setup()
+    {
+        $this->charge_divider = $this->__charge_divider();
+        $this->crossover = $this->__crossover();
+        $this->flatline_plus_charge = $this->__flatline_plus_charge();
+        $this->flatline = $this->__flatline();
+    }
+
+    private function __charge_divider()
+    {
+        return floatval(1 - $this->percentage);
+    }
+
+    private function __crossover()
+    {
+        return ceil(($this->crossover_total * $this->charge_divider) - $this->additional_charge);
+    }
+
+    private function __flatline_plus_charge()
+    {
+        return floor(($this->cap - $this->additional_charge) / $this->percentage);
+    }
+
+    private function __flatline()
+    {
+        return $this->flatline_plus_charge - $this->cap;
     }
-    $amount += $charge;
-    return $amount;
-    // if ($amountinkobo > KKD_PFF_PAYSTACK_FLATLINE_AMOUNT)
-    //     return ($amountinkobo + KKD_PFF_PAYSTACK_LOCAL_CAP)/100;
-    // elseif ($amountinkobo > KKD_PFF_PAYSTACK_CROSSOVER_AMOUNT)
-    //     return (intval(($amountinkobo + KKD_PFF_PAYSTACK_ADDITIONAL_CHARGE) / KKD_PFF_PAYSTACK_CHARGE_DIVIDER))/100;
-    // else
-    //     return (intval($amountinkobo / KKD_PFF_PAYSTACK_CHARGE_DIVIDER))/100;
+
+    public function add_for_kobo($amountinkobo)
+    {
+        if ($amountinkobo > $this->flatline) {
+            return $amountinkobo + $this->cap;
+        } elseif ($amountinkobo > $this->crossover) {
+            return ceil(($amountinkobo + $this->additional_charge) / $this->charge_divider);
+        } else {
+            return ceil($amountinkobo / $this->charge_divider);
+        }
+    }
+
+    public function add_for_ngn($amountinngn)
+    {
+        return $this->add_for_kobo(ceil($amountinngn * 100)) / 100;
+    }
+}
+
+function kkd_pff_paystack_add_paystack_charge($amount)
+{
+    $feeSettings = Kkd_Pff_Paystack_Public::fetchFeeSettings();
+    $pc = new PaystackCharge(
```

Comment:
Brother, you didn't update it here. 

```suggestion
    $pc = new Kkd_Pff_Paystack_PaystackCharge(
```

Reply:
Fixed